internal/controller/devicediscovery_controller.go

```diff
@@ -63,6 +65,28 @@ func setInitialsConditionsForDevice(device *v1alpha1.NicDevice) {
 	meta.SetStatusCondition(&device.Status.Conditions, condition)
 }
 
+func setFwConfigConditionsForDevice(device *v1alpha1.NicDevice) {
+	currentFirmware := helper.NicFirmwareMap[device.Status.FirmwareVersion]
+	recommendedFirmware := helper.NicFirmwareMap[device.Status.Type]
+	if currentFirmware == recommendedFirmware {
```

Comment:
@adrianchiris do we need to have '>=' here?

Reply:
Hi @e0ne following our discussion,

we should try to get mofed version from the system then check against recommended firmware version for the specific mofed version.

how we get the set of firmware versions recommended for  a mofed version should be queriable from mofed scripts or config map if we have no choice.

Reply:
until mofed scripts implement this feature, let's use configmap